.github/workflows/go-ci.yml

```diff
@@ -12,6 +12,7 @@ jobs:
         with:
           go-version: ^1
       - run: go test ./...
+      - run: go test -race ./... -update
```

Comment:
`-update` should not be ran in the CI pipeline.

Reply:
Agree that updating the golden files shouldn't happen as part of CI, though that's not the intention here.

The race condition happens when the `-update` flag is used only, and only testing without the `-update` flag doesn't exercise that behaviour through tests. I believe the behaviour when `-update` is used is that if any golden file needs to be updated because it doesn't match, the test run will fail, which is desirable here.

Reply:
Ahh I see, that makes sense.

Reply:
Perhaps add a comment indicating this is to test `-update` functionality does not have any lurking race conditions.

Reply:
Absolutely, a clarifying comment will help make the reasoning more obvious 👍

Reply:
I've also got another issue (and PR) to raise for a problem with `-update` when subtests with different parents have the same name. This relies on there being testing with the `-update` flag enabled to verify its behaviour as well, so will build on to of this change.

Reply:
Clarifying comment has been added

Reply:
The way to fix that will be to remove names as a requirement entirely: https://github.com/hexops/autogold/issues/11

I haven't gotten around to it yet, but should be pretty straightforward and very nice if we can just remove the names and use callstack line number info instead.